### DISCUSSION TITLE
Dedicated Feed Formats for Different Marketplaces

### DISCUSSION THREAD

**Discussion by fsyntax** [fsyntax/medusa-plugin-feeds#2] [General]
# Discussion: Dedicated Feed Formats for Different Marketplaces

## Context

The `medusa-plugin-feeds` plugin currently exposes two generic endpoints for fetching feeds in XML and JSON formats (`/feed/xml` and `/feed/json`). The question at hand is whether we should create dedicated feed formats tailored for specific marketplaces like Google Shopping, Facebook, and others that accept XML or JSON feeds.

## Pros

1. **Optimized Data Structure**: Each marketplace has its own set of requirements and preferences for feed data. Creating dedicated formats would allow us to optimize the data structure for each platform.
2. **Ease of Use**: Users wouldn't have to manually adjust the feed to meet the requirements of each marketplace.
3. **Feature Support**: Some marketplaces offer features that are not supported in a generic feed. Dedicated formats would allow us to take advantage of these features.

## Cons

1. **Maintenance**: More formats mean more code to maintain.
2. **Complexity**: Introducing multiple formats could make the plugin more complex to use.
3. **Resource Intensive**: Developing and maintaining multiple dedicated feeds could be resource-intensive.

## Open Questions

1. Should we prioritize certain marketplaces over others?
2. What would be the technical challenges involved in implementing dedicated formats?
3. Would this feature complicate the plugin's setup process for new users?

I'd love to hear your thoughts on this. Should we go ahead and develop dedicated feed formats for specific marketplaces, or should we stick with the generic XML and JSON formats?

**Comment by dwene** [comment]
Love the idea behind this project! We've had a client ask to start selling on Instagram, so this would be helpful for that purpose.

I would personally love to have specific ones for Google and Meta (facebook) at least. People can contribute other specific ones as needed, but I'd argue the biggest players out there right now are Meta and Google Shopping.

**Reply by fsyntax** [reply]
Thanks for your reply @dwene ! Thats basically exactly what I was aiming for and most probably going to work out first. 
Having dedicated feeds would allow for more flexibility, especially when paired with Medusas sales channel 👍

**Comment by srindom** [comment]
Would be cool if you could map Medusa entity fields to user-defined formats, then people can use it for all kinds of places.
